compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -65,12 +65,42 @@ static std::string getLinalgOpLoopRanges(linalg::LinalgOp op) {
   return outputString;
 }
 
+static std::string operandTypeToString(OpOperand *opOperand) {
+  auto operandType = opOperand->get().getType();
+  std::string outputString;
+  llvm::raw_string_ostream sstream(outputString);
+  if (auto shapedType = dyn_cast<ShapedType>(operandType)) {
+    shapedType.getElementType().print(sstream);
+  } else {
+    operandType.print(sstream);
+  }
+  return outputString;
+}
+
+// Returns a string like "f32xf32" representing a linalg op's types for each
+// operands.
```

Comment:
In order to keep the summaries short, how about collapsing "f32xf32xf32" down to just "f32" if all the types match?

What sorts of cases are motivating you to add the types to the summaries? For me, I'd probably want to know if a dispatch was mostly working in i8 vs i32 vs f32, and I wouldn't need the full list of types to see that.

Reply:
The use case is precisely that I need to know the datatype with the dispatch shapes. In this case, this is for profiling and tuning. 

I can make it so that only one type is printed if they all match. I am slightly worried this may be confusing between ops that print one or more types without an obvious correlation to operands.